src/applications/disability-benefits/all-claims/tests/utils/clean-toxic-exposure-data.unit.spec.jsx

```diff
@@ -0,0 +1,1466 @@
+          herbicide: { vietnam: true },
+        },
+      };
+
+      const formDataCopy = cloneDeep(originalFormData);
+
+      cleanToxicExposureData(originalFormData);
+
+      // Original data should remain unchanged
+      expect(originalFormData).to.deep.equal(formDataCopy);
+    });
+  });
+
+  describe('when cleaning individual exposure details', () => {
+    it('should remove Gulf War 1990 details for unselected locations', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          gulfWar1990: {
+            bahrain: false,
+            iraq: true,
+          },
+          gulfWar1990Details: {
+            bahrain: {
+              startDate: '1990-01-01',
+              endDate: '1990-12-31',
+            },
+            iraq: {
+              startDate: '1991-01-01',
+              endDate: '1991-12-31',
+            },
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure.gulfWar1990Details).to.not.have.property(
+        'bahrain',
+      );
+      expect(result.toxicExposure.gulfWar1990Details).to.have.property('iraq');
+    });
+
+    it('should remove Gulf War 2001 details for unselected locations', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { cancer: true },
+          gulfWar2001: {
+            afghanistan: true,
+            yemen: false,
+          },
+          gulfWar2001Details: {
+            afghanistan: {
+              startDate: '2001-10-01',
+              endDate: '2002-10-01',
+            },
+            yemen: {
+              startDate: '2002-01-01',
+              endDate: '2003-01-01',
+            },
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure.gulfWar2001Details).to.have.property(
+        'afghanistan',
+      );
+      expect(result.toxicExposure.gulfWar2001Details).to.not.have.property(
+        'yemen',
+      );
+    });
+
+    it('should remove herbicide details for unselected locations', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: { bronchitis: true },
+          herbicide: {
+            vietnam: true,
+            cambodia: false,
+          },
+          herbicideDetails: {
+            vietnam: {
+              startDate: '1968-01-01',
+              endDate: '1970-01-01',
+            },
+            cambodia: {
+              startDate: '1969-01-01',
+              endDate: '1970-01-01',
+            },
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result.toxicExposure.herbicideDetails).to.have.property('vietnam');
+      expect(result.toxicExposure.herbicideDetails).to.not.have.property(
+        'cambodia',
+      );
+    });
+
+    it('should remove otherHerbicideLocations when description is empty', () => {
```

Comment:
Can you confirm that this and other deleted objects aren't expected to be present in the form data by vets-api or Lighthouse, regardless of whether they are empty?